File: 1538-maximum-points-you-can-obtain-from-cards/maximum-points-you-can-obtain-from-cards.cpp
class Solution {
public:
    int maxScore(vector<int>& cardPoints, int k) {
        int n=cardPoints.size();
        int lSum=0,rSum=0,maxPoints=0;
        for(int i=0;i<k;i++) lSum+=cardPoints[i];
        maxPoints=lSum;
        
        int j=n-1;
        for(int i=k-1;i>=0;i--)
        {
            lSum-=cardPoints[i];
            rSum+=cardPoints[j];
            j--;
            maxPoints=max(maxPoints,(lSum+rSum));
        }
        return maxPoints;
    }
};

File: 1538-maximum-points-you-can-obtain-from-cards/maximum-points-you-can-obtain-from-cards.java
class Solution {
    public int maxScore(int[] nums, int k) {
        int n = nums.length;

        int leftSum = 0,rightSum = 0;
        for(int i=0;i<k;i++) {
            leftSum += nums[i];
        }

        int maxSum = leftSum, rightIndex = n-1;
        for(int i=k-1;i>=0;i--) {
            leftSum -= nums[i];
            rightSum += nums[rightIndex];
            rightIndex--;

            maxSum = Math.max(maxSum,leftSum + rightSum);
        }
        return maxSum;
    }
}